result_report.py

```diff
@@ -36,5 +37,41 @@ def end_result(self, result):
         markdown_file = sys.argv[2]
     except IndexError:
         markdown_file = "report.md"
+    try:
+        endpoints = sys.argv[3]
+    except IndexError:
+        endpoints = ""
+    try:
+        username = sys.argv[4]
+    except IndexError:
+        username = ""
+    try:
+        password = sys.argv[5]
+    except IndexError:
+        password = ""
+
+    # Parse output file and build markdown report.
     result = ExecutionResult(output_file)
-    result.visit(ResultReport())
\ No newline at end of file
+    result.visit(ResultReport())
+
+    # If endpoints are provided, add the version of these to the begin of the markdown report
+    if len(endpoints) > 0:
+        endpoint_versions = {}
+
+        urls = endpoints.split(',')
+        for url in urls:
+            if len(username) > 0:
+                response = requests.get(url + '/version.json', auth=(username, password))
+            else:
+                response = requests.get(url + '/version.json')
+            endpoint_versions[url.replace('https://','')] = response.json()['version']
+        with open(markdown_file,'r') as contents:
+            save = contents.read()
+        with open(markdown_file,'w') as contents:
+            contents.write("Tested components:\n")
+            contents.write("| Component | Version |\n")
+            contents.write("| -- | -- |\n")
+            for endpoint, version in endpoint_versions.items():
+                contents.write("| " + endpoint + " | " + version + " |\n")
+            contents.write("\n")
+            contents.write(save) 
```

Comment:
It would be nice to make it possible to not supply endpoints.
```suggestion
    endpoint_versions = {}

    # If endpoints are provided, add the version of these to the begin of the markdown report
    if len(endpoints) > 0:
        urls = endpoints.split(',')
        for url in urls:
            if len(username) > 0:
                response = requests.get(url + '/version.json', auth=(username, password))
            else:
                response = requests.get(url + '/version.json')
            endpoint_versions[url.replace('https://','')] = response.json()['version']

    with open(markdown_file,'r') as contents:
        save = contents.read()
    with open(markdown_file,'w') as contents:
        contents.write("Tested components:\n")
        contents.write("| Component | Version |\n")
        contents.write("| -- | -- |\n")
        for endpoint, version in endpoint_versions.items():
            contents.write("| " + endpoint + " | " + version + " |\n")
        contents.write("\n")
        contents.write(save) 
```

Reply:
In the current setup you can omit any endpoints and it won't add the component version table. In your suggestion it still would draw an empty table. I don't think we'd want that tbh.

Reply:
Ah I thought this logic runs after the end_result. This can be closed I will open another reaction for it.